### PR TITLE
Add option to control swap disk with vdisk

### PR DESCRIPTION
**Pull request by jichenjc**
this add an option to control whether swap is created from vdisk or mdisk

### CODE REVIEW

**Comment by jichenjc** [comment]
@bjhuangr @dongyanyang please help to review, thanks

**Comment by bjhuangr** [comment]
You should run tox locally to re-generate sample config in doc, and push the doc change as well

**Comment by jichenjc** [comment]
@bjhuangr just updated, thanks

File: zvmsdk/api.py [hunk 1]
```diff
@@ -816,6 +816,17 @@ def guest_create(self, userid, vcpus, memory, disk_list=None,
                                   " ECKD:eckdpoolname or FBA:fbapoolname")
                         LOG.error(errmsg)
                         raise exception.SDKInvalidInputFormat(msg=errmsg)
+                else:
+                    # in this case, it's swap only, and we will check whether
+                    # no VDISK is allowed, if not allow, then return error
+                    if disk_pool is None and CONF.zvm.swap_force_mdisk:
```
Comment (bjhuangr) [inline zvmsdk/api.py:822]: one minor comment: better to move the if condition to line 790, to raise exception earlier

**Review by bjhuangr — APPROVED** [review]

**Comment by bjhuangr** [comment]
LGTM

**Comment by bjhuangr** [comment]
@jichenjc please fix the conflicts, thx

File: zvmsdk/api.py (hunk 1 above)
Comment (jichenjc) [reply zvmsdk/api.py:822]: prefer to put here as `disk_pool` get from line 805 
and all code here are about validation test

**Review by bjhuangr — APPROVED** [review]

**Pull request merged**